File: Course-Registration-System-main/python-(iteration_2)/src/Transcript.py
from Grade import Grade
from TranscriptSaver import TranscriptSaver 
class Transcript:
    def __init__(self, transcriptId, GPA, grades):
        self.__transcriptId = transcriptId
        self.__GPA = GPA
        self.__grades = grades
        
    def getGrades(self):
        return self.__grades

    def setGrades(self, grades):
        self.__grades = grades

    def getId(self):
        return self.__transcriptId

    def setId(self, transcriptId):
        self.__transcriptId = transcriptId

    def getGPA(self):
        return self.__GPA

    def setGPA(self, GPA):
        self.__GPA = GPA

    def save(self):
        saver = TranscriptSaver()
        saver.save(self)
        pass

    def viewTranscript(self):
        transcriptInfo = []
        if self.__grades is not None:
            for grade in self.__grades:
                temp = ("\nCourse Code: " + str(grade.getCourseCode()) +
                        "\nCourse Name: " + grade.getCourseName() +
                        "\nCredit: " + str(grade.getCredit()) +
                        "\nNumeric Grade: " +str( grade.getNumericGrade()) +
                        "\nLetter Grade: " + grade.getLetterGrade() +
                        "\n--------------------------")
                transcriptInfo.append(temp)
        return transcriptInfo

File: Course-Registration-System-main/java-(iteration_1)/src/Transcript.java
import java.util.ArrayList;

//Represents a student's academic transcript.

public class Transcript  {
    // Identifier for the transcript
    private int id;
    // GPA of the student
    private double GPA;
    // List of grades associated with the transcript
    private ArrayList<Grade> grades;

    // Constructs a Transcript object.
    public Transcript(int id, double GPA, ArrayList<Grade> grades) {
        this.id = id;
        this.GPA = GPA;
        this.grades = grades;
    }

    // Default constructor for the Transcript class.
    public Transcript() {
    }

    // Getter and setter methods for each property.
    public ArrayList<Grade> getGrades() {
        return grades;
    }

    public void setGrades(ArrayList<Grade> grades) {
        this.grades = grades;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public double getGPA() {
        return GPA;
    }

    public void setGPA(double GPA) {
        this.GPA = GPA;
    }

    public void save() {
        // implementation
    }

    // Displays the contents of the transcript if available, otherwise indicates an
    // empty transcript.
    public String[] viewTranscript() {
        String[] transcriptInfo = new String[100];
        if (grades != null) {
            for (int i = 0; i < grades.size(); i++) {
                String temp = ("\nCourse Code: " + grades.get(i).getCourseCode() + "\nCourse Name: "
                        + grades.get(i).getCourseName() + "\nCredit: " + grades.get(i).getCredit() +
                        "\nNumeric Grade: " + grades.get(i).getNumericGrade() + "\nLetter Grade: "
                        + grades.get(i).getLetterGrade() + "\n--------------------------");
                transcriptInfo[i] = temp;
            }
        }
        return transcriptInfo;
    }
}
